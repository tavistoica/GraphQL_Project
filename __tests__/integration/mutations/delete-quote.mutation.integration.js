const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server");
const {
  database: { url },
} = require("../../../src/core/config/test.config");
const { Quote } = require("../../../src/entities/quote/quote.entity").default;
const { Server } = require("../../../src/bootstrap/server.bootstrap");
const { Database } = require("../../../src/bootstrap/database.bootstrap");

const server = new Server();
const database = new Database();

const quoteID = "123";

function getMutationCorrect() {
  return gql`
    mutation DeleteQuotes {
      deleteQuote(id: "123") {
        id
      }
    }
  `;
}

function getMutationNonExistentId() {
  return gql`
    mutation DeleteQuotes {
      deleteQuote(id: "456") {
        id
      }
    }
  `;
}

function createQuote(quote) {
  const newQuote = new Quote(quote);
  return newQuote.save();
}

async function createContext() {
  await createQuote({
    id: quoteID,
    phrase:
      "If you can't explain it to a six year old, you don't understand it yourself.",
    quoter: "Albert Einstein",
  });
  return {
    mutationCorrect: getMutationCorrect(),
    mutationNonExistentId: getMutationNonExistentId(),
  };
}

beforeEach(async () => {
  await database.bootstrap({ url });
  server.bootstrap();
});
afterEach(async () => {
  await database.reset();
  await server.close();
});

it("deletes a quote", async () => {
  const { mutationCorrect } = await createContext();
  const { mutate } = createTestClient(server.apolloClient);
  const res = await mutate({
    mutation: mutationCorrect,
  });
  expect(res).toMatchSnapshot();
});

it("checks the delete behavior when quote does not exist", async () => {
  const { mutationNonExistentId } = await createContext();
  const { mutate } = createTestClient(server.apolloClient);
  const res = await mutate({
    mutation: mutationNonExistentId,
  });
  expect(res).toMatchSnapshot();
});
