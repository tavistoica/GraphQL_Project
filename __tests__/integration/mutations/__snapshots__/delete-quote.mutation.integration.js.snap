// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the delete behavior when quote does not exist 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Quote does not exist in the database.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`deletes a quote 1`] = `
Object {
  "data": Object {
    "deleteQuote": Object {
      "id": "123",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
